server/src/main/java/com/paypal/selion/grid/servlets/transfer/DefaultManagedArtifact.java

```diff
@@ -48,29 +47,95 @@
 
     private static final String HTTP_CONTENT_TYPE = "application/zip";
 
-    private String filePath = null;
-
     private File artifactFile = null;
 
     private String artifactName = null;
 
-    private String folderName = null;
+    private String subFolderName = null;
 
-    private String parentFolderName = null;
+    private String uuidFolderName = null;
 
     private byte[] contents = null;
 
-    private final long timeToLiveInMillis;
+    private static final long timeToLiveInMillis = ConfigParser.parse().getLong(EXPIRY_CONFIG_PROPERTY);
+
+    private String requestPathInfo;
```

Comment:
Will do. It's the requested path to the artifact at download time. It is not the file system absolute path.